### ISSUE TITLE
User settings are overridden unexpectedly 

### ISSUE THREAD

**Issue by xieve** [Konfekt/vim-formatprgs#1]
Hi! I'd really like to use your plugin in my setup, but currently, it overrides some of the settings I have explicitly set globally:

<details>

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/c.vim#L8-L8

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/cmake.vim#L8-L10

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/cpp.vim#L8-L8

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/java.vim#L8-L8

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/javascript.vim#L3-L3

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/lua.vim#L8-L8

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/mail.vim#L9-L9

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/sh.vim#L1-L1

https://github.com/Konfekt/vim-formatprgs/blob/70346c16c8fc904a95b3c2d882fc542f61648acd/ftplugin/zsh.vim#L1-L5
</details>

I feel like these defaults are a little out-of-scope for this project. Maybe they could be moved to another plugin?

I have considered that I could explicitly re-set these either per filetype or globally via `setlocal`, but neither solution would satisfy me. Per-filetype just seems like unnecessary/redundant config, and using `setlocal` globally is quite dirty and would complicate it if I wanted to make my own per-filetype settings.

**Comment by Konfekt** [comment]
Yes, you're right. They are remnants of copy-pasting from my own copy-pasted config. How about opening a PR with https://github.com/xieve/vim-formatprgs/commit/552fefb006b97a757822c658e3b4c84fbc513471 that I'd gladly merge?
